### PR TITLE
limelight fixes

### PR DESCRIPTION
**Pull request by reslip1**

### CODE REVIEW

File: src/main/cpp/DragonVision/DragonLimelight.cpp [hunk 1]
```diff
@@ -454,10 +463,16 @@ std::optional<units::length::inch_t> DragonLimelight::EstimateTargetZDistance()
...
+    if (optionalTargetXDistance)
     {
-        units::length::inch_t targetXoffset_RF_inch = EstimateTargetXDistance().value() + GetMountingXOffset(); ///< the offset is negative if the limelight is behind the center of the robot
-
+        units::length::inch_t targetXoffset_RF_inch{0.0};
+        try {
+            targetXoffset_RF_inch = optionalTargetXDistance.value() + GetMountingXOffset(); ///< the offset is negative if the limelight is behind the center of the robot
+            //Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, std::string("LL"), std::string("EstimateTargetXDistance_RelToRobotCoords"), targetXoffset_RF_inch.to<double>());
+        } catch (std::bad_optional_access &e) {
+            //Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, std::string("LL"), std::string("EstimateTargetXDistance_RelToRobotCoords-exception"), e.what());
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonLimelight.cpp:474]: If we are going to keep this, can we return std::nullopt when we catch an exception

File: src/main/cpp/DragonVision/DragonLimelight.cpp [hunk 2]
```diff
@@ -468,7 +483,7 @@ std::optional<units::length::inch_t> DragonLimelight::EstimateTargetYDistance_Re
     if (EstimateTargetYDistance())
     {
         units::length::inch_t targetYoffset_RF_inch = EstimateTargetYDistance().value() + GetMountingYOffset(); ///< the offset is positive if the limelight is to the left of the center of the robot
-
+        //Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, std::string("LL"), std::string("EstimateTargetYDistance_RelToRobotCoords"), targetYoffset_RF_inch.to<double>());
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonLimelight.cpp:486]: Remove debug logging

File: src/main/cpp/DragonVision/DragonLimelight.cpp [hunk 3]
```diff
@@ -480,7 +495,7 @@ std::optional<units::length::inch_t> DragonLimelight::EstimateTargetZDistance_Re
     if (EstimateTargetZDistance())
     {
         units::length::inch_t targetZoffset_RF_inch = EstimateTargetZDistance().value() + GetMountingZOffset(); ///< the offset is positive if the limelight is above the center of the robot
-
+        //Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, std::string("LL"), std::string("EstimateTargetZDistance_RelToRobotCoords"), targetZoffset_RF_inch.to<double>());
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonLimelight.cpp:498]: Remove debug logging

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 4]
```diff
@@ -25,6 +25,7 @@
 #include "DragonVision/DragonPhotonCam.h"
 #include "utils/FMSData.h"
 #include "DragonVision/DragonVisionStructLogger.h"
+#include "utils/logging/Logger.h"
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonVision.cpp:28]: Remove debug logging

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 5]
```diff
@@ -220,42 +221,67 @@ std::optional<VisionData> DragonVision::GetDataToNearestAprilTag(RobotElementNam
 std::optional<VisionData> DragonVision::GetVisionDataFromNote(VISION_ELEMENT element)
 {
 	DragonCamera *selectedCam = nullptr;
+	//std::string loggerName = "getVisionDataFromNote";
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonVision.cpp:224]: Remove comments

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 6]
```diff
@@ -220,42 +221,67 @@ std::optional<VisionData> DragonVision::GetDataToNearestAprilTag(RobotElementNam
...
 		selectedCam = m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::PINTAKE];
+		//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("placer_note"));
 		break;
 	case VISION_ELEMENT::LAUNCHER_NOTE:
 		selectedCam = m_dragonCameraMap[RobotElementNames::CAMERA_USAGE::LINTAKE];
+		//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("launcher_note"));
 		break;
 	case VISION_ELEMENT::NOTE:
 	{
+		//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("note"));
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonVision.cpp:238]: Remove comments above

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 7]
```diff
@@ -220,42 +221,67 @@ std::optional<VisionData> DragonVision::GetDataToNearestAprilTag(RobotElementNam
...
 		}
 
 		if (!lintakeHasDetection && !pintakeHasDetection)
 		{
+			//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("no detect"));
 			return std::nullopt;
 		}
 		else if (lintakeHasDetection && pintakeHasDetection)
 		{
+			//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("both detect"));
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonVision.cpp:263]: Remove comments above

File: src/main/cpp/DragonVision/DragonVision.cpp [hunk 8]
```diff
@@ -275,11 +301,14 @@ std::optional<VisionData> DragonVision::GetVisionDataFromNote(VISION_ELEMENT ele
...
 	}
 
+	//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("selectedCam"));
 	// double check selectedCam is not nullptr
 	if (selectedCam != nullptr)
 	{
+		//Logger::GetLogger()->LogData(LOGGER_LEVEL::PRINT, loggerName, std::string("VisionElement"), std::string("transforming"));
 		// create translation using 3 estimated distances
 		frc::Translation3d translationToNote = frc::Translation3d(selectedCam->EstimateTargetXDistance_RelToRobotCoords().value(), selectedCam->EstimateTargetYDistance_RelToRobotCoords().value(), selectedCam->EstimateTargetZDistance_RelToRobotCoords().value());
+		//DragonVisionStructLogger().logTranslation3d(std::string("translationToNote"),translationToNote);
```
Comment (nickfrisch) [inline src/main/cpp/DragonVision/DragonVision.cpp:311]: Remove comments above

File: src/main/cpp/Robot.cpp [hunk 9]
```diff
@@ -129,6 +129,7 @@ void Robot::TeleopPeriodic()
         m_holonomic->Run();
     }
     PeriodicLooper::GetInstance()->TeleopRunCurrentState();
+    DragonVision::GetDragonVision()->testAndLogVisionData();
```
Comment (nickfrisch) [inline src/main/cpp/Robot.cpp:132]: Remove

**Review by nickfrisch — CHANGES_REQUESTED** [review]

File: src/main/cpp/DragonVision/DragonVision.cpp (hunk 4 above)
Comment (reslip1) [reply src/main/cpp/DragonVision/DragonVision.cpp:28]: done. the only logging left is in that test function that's never called

File: src/main/cpp/DragonVision/DragonLimelight.cpp (hunk 1 above)
Comment (reslip1) [reply src/main/cpp/DragonVision/DragonLimelight.cpp]: i removed that as I was able to track it down

**Comment by reslip1** [comment]
no longer needed

**Pull request closed without merging**